apps/ledger-live-mobile/src/screens/WalletCentricAsset/AssetRow.tsx

```diff
@@ -53,12 +55,20 @@ const AssetRow = ({
     });
   }, [currency, navigation]);
 
+  /**
+   * Avoid passing a new object to AssetRowLayout if the value didn't actually
+   * change.
+   * Not a small optimisation as that component can take several milliseconds to
+   * render, and it's meant to be rendered in a list.
+   *  */
+  const balance = useMemo(() => BigNumber(asset.amount), [asset.amount]);
+
   return (
     <AssetRowLayout
       onPress={onAssetPress}
       currency={currency}
       currencyUnit={unit}
-      balance={new BigNumber(asset.amount)}
```

Comment:
not related to the PR, but I'm thinking about why we convert this number to BigNumber in the first place, as if it's a Number we already lost the BigNumber part, and if not then it's useless to convert :sneezing_face:

Reply:
Yeah legacy logic I guess 😅